documentation/c09-concepts.sil

```diff
@@ -2,71 +2,152 @@
 \include[src=documentation/macros.sil]
 \chapter{The Nitty Gritty}
 
-\noindent{}We are finally at the bottom of our delve into SILE’s commands
-and settings. Here are the basic building blocks out of which all of the
-other operations in SILE are created.
+We are finally at the bottom of our delve into SILE’s commands and settings.
+Here are the basic building blocks out of which all of the other operations in SILE are created.
 
-\note{At this point, it is expected that
-you are a class designer, and will be able to follow the details of how
-SILE implements these commands and features; we will also explain how to
-interact with these components at the Lua level.}
+\note{At this point, it is expected that you are a class or package designer, and will be able to follow the details of how SILE implements these commands and features;
+we will also explain how to interact with these components at the Lua level.}
+
+\section{Measurements and lengths}
+
+Before dabbling into more advanced topics, let’s introduce “measurements” and “lengths” in SILE,
+the two available Lua constructs for representing dimensions.
+
+Measurements are specified in terms of \code{SILE.measurement} objects.
+It is a basic construct with an amount and a unit. Let us illustrate two common ways for creating such an object in Lua (from a string, with same syntax as in command parameters; or from a Lua table).
+
+\begin[type=autodoc:codeblock]{raw}
+local m1 = SILE.measurement("10pt")
+local m2 = SILE.measurement({ amount = 10, unit = "pt" })
+\end{raw}
+
+SILE also provides a more advanced construct specified in terms of \code{SILE.length} objects;
+these are “three-dimensional” dimensions, in that they consist in a base measurement plus stretch and shrink measurements.
+They are therefore composed of three \code{SILE.measurement}.
+
+\begin[type=autodoc:codeblock]{raw}
+local l1 = SILE.length("10pt plus 2pt minus 1pt")
+local l2 = SILE.length({ length = "1Opt", stretch = "2pt", shrink = "1pt" })
+\end{raw}
+
+Both of these are used for various purposes.
+In many cases, they are nearly interchangeable.
+Casting from one to the other is straightforward:
+casting a length to a measurement returns just the base measurement and discards the stretch and shrink properties;
+casting a measurement to a length sets its stretch and shrink properties to zero.
+
+\begin[type=autodoc:codeblock]{raw}
+local l3 = SILE.length(SILE.measurement("10pt")) -- 10pt, without strech and shrink
```

Comment:
"without strech ..." --> "without stretch'. It seems I cannot get this word right, despite knowing I do the same mistake over and over :)

Reply:
Fixed in a separate commit: a few other occurrence (not mine only, pfew).